In [1]:
import pandas as pd
from math import pow
import numpy as np

Problem Statement 1:<br>
Is gender independent of education level? A random sample of 395 people were
surveyed and each person was asked to report the highest education level they
obtained. The data that resulted from the survey is summarized in the following table:<br>

       High School   Bachelors   Masters  Ph.d. Total<br>
Female  60            54          46       41    201<br>
Male    40            44          53       57    194<br>
Total  100            98          99       98    395<br>
Question: Are gender and education level dependent at 5% level of significance? In
other words, given the data collected above, is there a relationship between the
gender of an individual and the level of education that they have obtained?

We are going to use Chi squared test

In [2]:
n=395

In [3]:
E1= round((100*201)/n,2)
E1

50.89

In [4]:
df= pd.DataFrame([[60,54,46,41,201],[40,44,53,57,194],[100,98,99,98,395]], index=['Female','Male','Total'], columns=['High School','Bachelors','Masters','Ph.d.','Total'])
df

High School  Bachelors  Masters  Ph.d.  Total
Female           60         54       46     41    201
Male             40         44       53     57    194
Total           100         98       99     98    395

In [5]:
O=list(map(float,[60,54,46,41,40,44,53,57]))

E= (total of rows*total of column)/ sample size

In [6]:
E=[]
n=395
s=0
t1=[100,98,99,98]
t2=[201,194]
for i in range(0,2):
    for j in range(0,4):
        s= round((t1[j]*t2[i])/n,3)
        E.append(s)
E

[50.886, 49.868, 50.377, 49.868, 49.114, 48.132, 48.623, 48.132]

chi<sup>2</sup>= sum(O-E)<sup>2</sup>/E

In [7]:
chi=0
for i in range(len(O)):
    chi += (pow((O[i]-E[i]),2)/E[i])
chi= round(chi,3)
chi

8.006

The critical value of chi2  with 3 degree of freedom is 7.815. Since 8.006 > 7.815, we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.

Problem Statement 2:<br>
Using the following data, perform a oneway analysis of variance using α=.05. Write
up the results in APA format.<br>
[Group1: 51, 45, 33, 45, 67]<br>
[Group2: 23, 43, 23, 43, 45]<br>
[Group3: 56, 76, 74, 87, 56]

In [8]:
df1=pd.DataFrame([[51, 45, 33, 45, 67],[23, 43, 23, 43, 45],[56, 76, 74, 87, 56]], index=['Group1','Group2','Group3'])
df1=df1.T
df1

Group1  Group2  Group3
0      51      23      56
1      45      43      76
2      33      23      74
3      45      43      87
4      67      45      56

In [9]:
l=['Group1','Group2','Group3']
m=[]
print("Mean:")
for i in l:
    m.append(df1[i].mean())
m

Mean:


[48.2, 35.4, 69.8]

In [10]:
df.shape

(3, 5)

In [11]:
sd1=df1['Group1']- m[0]
sd2=df1['Group2']- m[1]
sd3=df1['Group3']- m[2]

In [12]:
ssd1= sd1**2
ssd2= sd2**2
ssd3= sd3**2

In [13]:
ssd1

0      7.84
1     10.24
2    231.04
3     10.24
4    353.44
Name: Group1, dtype: float64

In [14]:
var1= sum(ssd1)/(5-1)
var2= sum(ssd2)/(5-1)
var3= sum(ssd3)/(5-1)

In [15]:
MSerror= (var1+var2+var3)/3
MSerror

155.06666666666666

In [16]:
SSerror= (MSerror)*(15-3)
SSerror

1860.8

In [17]:
x= round(sum(m)/3,1)
x

51.1

In [18]:
gm=[]
for i in range(0,3):
    gm.append(round(m[i]-x,1))
gm

[-2.9, -15.7, 18.7]

In [19]:
sgm=[]
for i in range(0,3):
    sgm.append(gm[i]**2)
sgm

[8.41, 246.48999999999998, 349.69]

In [20]:
ssmean= sum(sgm)
ssmean

604.5899999999999

In [21]:
var2= ssmean/(3-1)
var2

302.29499999999996

In [22]:
MS= (var2)*5
MS

1511.475

In [23]:
df= 3-1
df

2

In [24]:
ss= MS*df
ss

3022.95

In [25]:
F= round(MS/MSerror,2)
F

9.75

f value is 3.89 which is smaller than 9.75. So we reject the null hypothesis.

Problem Statement 3:<br>
Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25.<br>
For 10, 20, 30, 40, 50:

In [26]:
l1=np.array([10, 20, 30, 40, 50])
l2=np.array([5,10,15, 20, 25])
m1= l1.mean()
m1

30.0

In [27]:
sd1= l1.std()
sd1

14.142135623730951

In [28]:
var1= (sd1)**2
var1

200.00000000000003

In [29]:
m2= l2.mean()
m2

15.0

In [30]:
sd2= l2.std()
sd2

7.0710678118654755

In [31]:
var2= (sd2)**2
var2

50.00000000000001

In [32]:
f_test= var1/var2
f_test

4.0